Downsample/Downsample.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+Import a ZDF point cloud and downsample it.
+*/
+
+#include <Zivid/CloudVisualizer.h>
+#include <Zivid/Zivid.h>
+#include <Eigen/Dense>
+#include <Eigen/Core>
+
+#include <iostream>
+Eigen::MatrixXf sumline(Eigen::MatrixXf&, int, int);
+Eigen::MatrixXf gridsum(Eigen::MatrixXf&, int);
+float myround(float); 
+float isnanmask(float);
+float nantozero(float);
+Zivid::PointCloud downsample(Zivid::PointCloud&, int);
+
+int main()
+{
+	try
+	{
+		Zivid::Application zivid;
+
+		std::string Filename = "Zivid3D.zdf";
+		std::cout << "Reading " << Filename << " point cloud" << std::endl;
+		Zivid::Frame frame = Zivid::Frame(Filename);
+
+		auto pointCloud = frame.getPointCloud();
+		auto downSamplingFactor = 4;
+
+		Zivid::PointCloud pointCloudDownsampled(pointCloud.height()/downSamplingFactor, pointCloud.width()/downSamplingFactor);
+
+		pointCloudDownsampled = downsample(pointCloud, downSamplingFactor);
+
+		std::cout << "Setting up visualization" << std::endl;
+		{
+			Zivid::CloudVisualizer vis;
+			zivid.setDefaultComputeDevice(vis.computeDevice());
+
+			std::cout << "Displaying the original point cloud" << std::endl;
+			vis.showMaximized();
+			vis.show(pointCloud);
+			vis.resetToFit();
+
+			std::cout << "Running the visualizer. Blocking until the window closes" << std::endl;
+			vis.run();
+		}
+		{
+			Zivid::CloudVisualizer vis;
+			zivid.setDefaultComputeDevice(vis.computeDevice());
+
+			std::cout << "Displaying the downsampled point cloud" << std::endl;
+			vis.showMaximized();
+			vis.show(pointCloudDownsampled);
+			vis.resetToFit();
+
+			std::cout << "Running the visualizer. Blocking until the window closes" << std::endl;
+			vis.run();
+		}
+	}
+
+	catch (const std::exception &e)
+	{
+		std::cerr << "Error: " << Zivid::toString(e) << std::endl;
+		return EXIT_FAILURE;
+	}
+}
+
+Eigen::MatrixXf sumline(Eigen::MatrixXf& matrix, int downSamplingFactor)
```

Comment:
In general I prefer to have functions in "reading order". So it's good that you have `main` on top, as that's the entry point. Then I'd continue with `downsample`, as that's the one being called from `main`. Then I'd continue with the functions called by `downsample`, and then the ones called from those again etc.